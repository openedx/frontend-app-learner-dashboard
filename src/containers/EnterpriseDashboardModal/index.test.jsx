import { shallow } from '@edx/react-unit-test-utils';
import EnterpriseDashboard from '.';

import useEnterpriseDashboardHook from './hooks';

jest.mock('./hooks', () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe('EnterpriseDashboard', () => {
  test('snapshot', () => {
    const hookData = {
      dashboard: { label: 'edX, Inc.', url: '/edx-dashboard' },
      showDialog: false,
      handleClose: jest.fn().mockName('useEnterpriseDashboardHook.handleClose'),
      handleCTAClick: jest.fn().mockName('useEnterpriseDashboardHook.handleCTAClick'),
      handleEscape: jest.fn().mockName('useEnterpriseDashboardHook.handleEscape'),
    };
    useEnterpriseDashboardHook.mockReturnValueOnce({ ...hookData });
    const el = shallow(<EnterpriseDashboard />);
    expect(el.snapshot).toMatchSnapshot();
  });
  test('empty snapshot', () => {
    useEnterpriseDashboardHook.mockReturnValueOnce({});
    const el = shallow(<EnterpriseDashboard />);
    expect(el.snapshot).toMatchSnapshot();
  });
});
