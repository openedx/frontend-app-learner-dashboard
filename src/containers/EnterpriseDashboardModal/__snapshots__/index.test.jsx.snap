// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterpriseDashboard empty snapshot 1`] = `null`;

exports[`EnterpriseDashboard snapshot 1`] = `
<ModalDialog
  hasCloseButton={false}
  onClose={[MockFunction useEnterpriseDashboardHook.handleEscape]}
  title=""
>
  <div
    className="bg-white p-3 rounded shadow"
    style={
      {
        "textAlign": "start",
      }
    }
  >
    <h4>
      You have access to the edX, Inc. dashboard
    </h4>
    <p>
      To access the courses available to you through edX, Inc., visit the edX, Inc. dashboard now.
    </p>
    <ActionRow>
      <Button
        onClick={[MockFunction useEnterpriseDashboardHook.handleClose]}
        variant="tertiary"
      >
        Dismiss
      </Button>
      <Button
        href="/edx-dashboard"
        onClick={[MockFunction useEnterpriseDashboardHook.handleCTAClick]}
        type="a"
      >
        Go to dashboard
      </Button>
    </ActionRow>
  </div>
</ModalDialog>
`;
